viewer/src/views/threejs/cad/shading.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import { createMaterials, Materials } from './materials';
+
+export interface Shading {
+  materials: Materials;
+  updateNodes: (treeIndices: number[]) => void;
+}
+
+export interface DefaultShadingOptions {
+  color: (treeIndex: number) => number[] | undefined;
+}
+
+export function createDefaultShading(options: DefaultShadingOptions) {
+  const materials = createMaterials();
+  const updateNodes = (treeIndices: number[]) => {
+    if (treeIndices.length === 0) {
+      return;
+    }
+    for (const treeIndex of treeIndices) {
+      const colorOrUndefined = options.color(treeIndex);
+      const color = colorOrUndefined ? colorOrUndefined : [0, 0, 0, 0];
+      materials.colorDataTexture.image.data[4 * treeIndex] = color[0];
+      materials.colorDataTexture.image.data[4 * treeIndex + 1] = color[1];
+      materials.colorDataTexture.image.data[4 * treeIndex + 2] = color[2];
+      materials.colorDataTexture.image.data[4 * treeIndex + 3] = color[3];
+    }
+    materials.colorDataTexture.needsUpdate = true;
```

Comment:
Should we always do this or try to be resilient to unnecessary calls to `updateNodes()` by comparing values in the loop? Not sure what's best...

Reply:
I see what you mean, but I think I prefer the simpler version for now, until we see the need for an optimization. Since `updateNodes()` already requires treeIndices to be specified, it should reduce the temptation to call it everywhere "for good measure", which is usually when it becomes necessary to do those types of checks. The current use case is when the user clicks something, and at that point the texture will very likely change.